### PR TITLE
LightsOff: create recipe

### PR DESCRIPTION
**Pull request by owenca**
Will update this recipe to reflect the latest upstream commit once this [PR](https://github.com/HaikuArchives/LightsOff/pull/1) is merged.

### CODE REVIEW

**Comment by owenca** [comment]
Updated the recipe to the latest upstream commit.

File: haiku-games/lightsoff/lightsoff-1.0.0.recipe [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:g++
+	cmd:make
+	"
+
+BUILD()
```
Comment (fbrosson) [inline haiku-games/lightsoff/lightsoff-1.0.0.recipe]:
```diff
 	"
 
+defineDebugInfoPackage lightsoff \
+	$appsDir/$APP
+
 BUILD()
```

Can you please add a debuginfo package?
Thanks!

Comment (owenca) [reply haiku-games/lightsoff/lightsoff-1.0.0.recipe]: Done. :)

**Comment by fbrosson** [comment]
According to the About box, the version is `v1.01 b1`. However, the rdef says:
* major  = 1,
* middle = 0,
* minor  = 0,

Hence this question: should we rename the recipe to `lightsoff-1.0.recipe` or shall we keep it with its current name, `lightsoff-1.0.0.recipe` ?
(I'm OK with both choices.)
I'm just asking this now, before the merge, in case someone thinks we should rename the recipe.

**Comment by owenca** [comment]
I don't have an opinion on this but prefer a shorter name in general. :)

**Comment by fbrosson** [comment]
Cool!
(I also prefer `lightsoff-1.0.recipe`.)
Let's go for the rename then :)

**Comment by fbrosson** [comment]
Hmm, I thought I had already reported a small issue I had found on x86_64, but it looks like I did not submit my comment... So here it is:

On x86_64, all 9 cpp files (AboutWindow.cpp App.cpp DrawButton.cpp Grid.cpp GridView.cpp MainWindow.cpp Preferences.cpp PuzzlePack.cpp TwoStateDrawButton.cpp) build fine.
But when we reach the next step, we hit this error:
```
cat Resources.rdef | cc -E -iquote./  -iquote./  -iquoteobjects/   -O0   -Woverloaded-virtual -funsigned-bitfields -Wwrite-strings - | grep -av '^#' | rc -I ./ -o "objects/Resources.rsrc" -
cc1: warning: command line option '-Woverloaded-virtual' is valid for C++/ObjC++ but not for C
```
I guess the problem can be fixed with a few changes in the Makefile.

Would you like to investigate this, so that we can mark x86_64 as tested if you find the fix? Thanks!

**Comment by owenca** [comment]
The cc1 warning shouldn't be a problem. Do you have the log after the cat command? Anyway, this is what I got for x86:
`cat Resources.rdef | cc -E -iquote./  -iquote./  -iquoteobjects.x86-cc5-release/   -O0   -Woverloaded-virtual -funsigned-bitfields -Wwrite-strings - | grep -av '^#' | rc -I ./ -o "objects.x86-cc5-release/Resources.rsrc" -
cc1: warning: command line option '-Woverloaded-virtual' is valid for C++/ObjC++ but not for C
cc -o "objects.x86-cc5-release/LightsOff"       objects.x86-cc5-release/AboutWindow.o objects.x86-cc5-release/App.o objects.x86-cc5-release/DrawButton.o objects.x86-cc5-release/Grid.o objects.x86-cc5-release/GridView.o objects.x86-cc5-release/MainWindow.o objects.x86-cc5-release/Preferences.o objects.x86-cc5-release/PuzzlePack.o objects.x86-cc5-release/TwoStateDrawButton.o  -Xlinker -soname=_APP_  -L./  -Lobjects.x86-cc5-release/    -ltranslation  -lbe  -lgame  -lroot 
objects.x86-cc5-release/AboutWindow.o: In function `AboutWindow::AboutWindow()':
AboutWindow.cpp:(.text+0xd0): undefined reference to `operator new(unsigned long)'
objects.x86-cc5-release/App.o: In function `MyApp::MyApp()':
App.cpp:(.text+0x87): undefined reference to `operator new(unsigned long)'
objects.x86-cc5-release/GridView.o: In function `GridView::GridView()':
GridView.cpp:(.text+0x2b8): undefined reference to `operator new(unsigned long)'
GridView.cpp:(.text+0x2ff): undefined reference to `operator new(unsigned long)'
GridView.cpp:(.text+0x36a): undefined reference to `operator new(unsigned long)'
objects.x86-cc5-release/GridView.o:GridView.cpp:(.text+0x3bf): more undefined references to `operator new(unsigned long)' follow
collect2: error: ld returned 1 exit status
/boot/system/develop/etc/makefile-engine:246: recipe for target 'objects.x86-cc5-release/LightsOff' failed
make: *** [objects.x86-cc5-release/LightsOff] Error 1
[x86] ~/LightsOff/src> 
`

**Comment by nysnatuss** [comment]
@owenca I had fixed the makefile to build on x86_64 at upstream (HaikuArchives/LightsOff#12) so it should be working now. @fbrosson Can you help confirm that it works on x86_64? :)

File: haiku-games/lightsoff/lightsoff-1.0.recipe [hunk 2]
```diff
@@ -0,0 +1,46 @@
+SUMMARY="A puzzle game based on the original handheld game Lights Out"
+DESCRIPTION="LightsOff is a redition of the addictive puzzle game Lights Out."
+APP="LightsOff"
+HOMEPAGE="https://github.com/HaikuArchives/$APP"
+COPYRIGHT="2005 DarkWyrm"
+LICENSE="MIT"
+REVISION="1"
+srcGitRev="c109cee2abd23d1370be780c3899607508fd837e"
```
Comment (fbrosson) [inline haiku-games/lightsoff/lightsoff-1.0.recipe]:
```diff
-srcGitRev="c109cee2abd23d1370be780c3899607508fd837e"
+srcGitRev="40b11f1f2f24a525d3b5c1a6336ec13aa3abbad7"
```

File: haiku-games/lightsoff/lightsoff-1.0.recipe [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+SOURCE_URI="$HOMEPAGE/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="0ffa282a76db1d0b50d8f8808d4eb41c2a4896fea052ef93771815efc066db86"
```
Comment (fbrosson) [inline haiku-games/lightsoff/lightsoff-1.0.recipe]:
```diff
-CHECKSUM_SHA256="0ffa282a76db1d0b50d8f8808d4eb41c2a4896fea052ef93771815efc066db86"
+CHECKSUM_SHA256="01f9fc079f45b88cc5b39c64c9a3866d435bd8a16fe6f296e48482716df80d78"
```

File: haiku-games/lightsoff/lightsoff-1.0.recipe [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+SOURCE_FILENAME="$APP-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="$APP-$srcGitRev"
+
+ARCHITECTURES="x86_gcc2 ?x86_64"
```
Comment (fbrosson) [inline haiku-games/lightsoff/lightsoff-1.0.recipe]:
```diff
-ARCHITECTURES="x86_gcc2 ?x86_64"
+ARCHITECTURES="x86_gcc2 x86_64"
```

**Comment by owenca** [comment]
@fbrosson I have updated the recipe. However, I had to take out `defineDebugInfoPackage` because it causes the grid/buttons of the game to not show, probably due to stripping `mime_db` from `data` in the package.

**Comment by owenca** [comment]
Also, the stripped debugging info was less than 10K bytes. Perhaps we will not strip it for this little game?

**Comment by fbrosson** [comment]
I think I know what is causing the buttons to not show.
Where is the haikuporter you are using? Is it the one in `/boot/system/bin/` ?
I have recently fixed a bug in haikuporter which was causing the resources to be dropped when the recipe created a debuginfo package. But there hasn't been a version bump yet, so the only way to call avoid the bug is to download haikuporter from git.
Could you try this:
```
cd /boot/home/
git clone https://github.com/haikuports/haikuporter.git
cd /boot/home/haikuports/haiku-games/lightsoff
/boot/home/haikuporter/haikuporter -S lightsoff
```
(Yes, I know. We don't need to go to the directory where the recipe is in order to call haikuporter.)
Thanks!

**Comment by owenca** [comment]
You were right! Everything works with the cloned `haikuporter`. I even got a source package this time. :)

**Pull request merged**